src/RA_Defs.h

```diff
@@ -218,47 +244,79 @@ const int SERVER_PING_DURATION = 2 * 60;
 #ifdef _DEBUG
 #ifndef RA_UTEST
 #undef ASSERT
-#define ASSERT( x ) assert( x )
+_CONSTANT_FN ASSERT(_In_ bool bExpr) noexcept { assert(bExpr); }
 #else
 #undef ASSERT
-#define ASSERT( x ) {}
+_CONSTANT_FN ASSERT(_UNUSED bool) noexcept {}
 #endif
 #else
 #undef ASSERT
-#define ASSERT( x ) {}
+_CONSTANT_FN ASSERT(_UNUSED bool) noexcept {}
 #endif
 
 #ifndef UNUSED
-#define UNUSED( x ) ( x );
+template<typename T>
+_CONSTANT_FN UNUSED(_UNUSED T) noexcept {}
 #endif
 
 namespace ra {
 
 _NODISCARD std::string Narrow(_In_ const std::wstring& wstr);
-_NODISCARD std::string Narrow(_Inout_ std::wstring&& wstr) noexcept;
+_NODISCARD std::string Narrow(_In_ std::wstring&& wstr) noexcept;
 _NODISCARD std::string Narrow(_In_z_ const wchar_t* wstr);
 _NODISCARD std::wstring Widen(_In_ const std::string& str);
-_NODISCARD std::wstring Widen(_Inout_ std::string&& str) noexcept;
+_NODISCARD std::wstring Widen(_In_ std::string&& str) noexcept;
 _NODISCARD std::wstring Widen(_In_z_ const char* str);
 
 //	No-ops to help convert:
 _NODISCARD std::wstring Widen(_In_z_ const wchar_t* wstr);
 _NODISCARD std::wstring Widen(_In_ const std::wstring& wstr);
+_NODISCARD std::wstring Widen(_In_ std::wstring&& wstr) noexcept;
 _NODISCARD std::string Narrow(_In_z_ const char* str);
-_NODISCARD std::string Narrow(_In_ const std::string& wstr);
+_NODISCARD std::string Narrow(_In_ const std::string& str);
+_NODISCARD std::string Narrow(_In_ std::string&& str) noexcept;
 _NODISCARD std::string ByteAddressToString(_In_ ra::ByteAddress nAddr);
 
-} // namespace ra
-
 
+template<typename CharT, class = std::enable_if_t<is_char_v<CharT>>>
+_NODISCARD inline auto NativeStr(_In_ const CharT* str)
+{
+    static_assert(!std::is_same_v<CharT, unsigned char>, "Conversion for unsigned strings is currently unsupported!");
+#if _MBCS
+    return Narrow(str);
+#elif _UNICODE
+    return Widen(str);
+#else
+#error Unknown character set detected! Only MultiByte and Unicode are supported!
+#endif // _MBCS
+}
```

Comment:
Basically made `NativeStr `into 3 function templates instead of 6 (including no-ops maybe 12?) regular functions. These either call the no-op or op function depending on what `CharT `is.